Civi/Install/Requirements.php

```diff
@@ -102,6 +102,17 @@ public function checkDatabase(array $db_config) {
   }
 
   /**
+   * Generates a mysql connection
+   *
+   * @param $db_confic array
+   * @return object mysqli connection
+   */
+  protected function connect($db_config) {
+    $conn = @mysqli_connect($db_config['host'], $db_config['username'], $db_config['password'], $db_config['database'], isset($db_config['port']) ? $db_config['port'] : NULL);
```

Comment:
Big +1 for moving this to a function.

On switching `host` vs `server`, I feel like the issue is that there's some inconsistency where different use-cases provide this field differently. (So fixing one use-case breaks the other.) Maybe use a fallback like

```php
if (!empty($db_config['host'])) $host = $db_config['host'];
elseif (!empty($db_config['server'])) $host = $db_config['server']
else $host = NULL;
```